src/lualib/ArrayFindIndex.ts

```diff
@@ -1,6 +1,7 @@
 function __TS__ArrayFindIndex<T>(
+    this: void,
     arr: T[],
-    callbackFn: (this: void, element: T, index?: number, array?: T[]) => boolean
+    callbackFn: (element: T, index?: number, array?: T[]) => boolean
```

Comment:
So this callback inherits the void context from the function definition? Maybe it's me, but this doesn't seem very intuitive to me.

Reply:
No, it's not inherited. I set the callback to take as self, so regularly defined functions can be passed in here. Otherwise any non-expression functions used as callbacks would need the `this: void` which would get annoying.